### ISSUE TITLE
Improve scaling quality of Windows direct3d renderer

### ISSUE THREAD

**Issue by rustyx** [Genymobile/scrcpy#1422]
On Windows, the default renderer is `direct3d` which is nice because it's much faster than `opengl`.
But unfortunately this renderer doesn't seem to support trilinear filtering as is evident from the log:

```
[server] INFO: Device: samsung SM-N910C (Android 6.0.1)
INFO: Renderer: direct3d
WARN: Trilinear filtering disabled (not an OpenGL renderer)
INFO: Initial texture: 1440x2560
```

Is there any chance of improving the jagged scaling with Direct3D? Tried forcing hardware-accelerated antialiasing via NVIDIA control panel to no avail.

**Environment**
 - OS: Windows 10 x64
 - scrcpy version: scrcpy 1.13                                                                                                                                                     
dependencies:
 - SDL 2.0.12
 - libavcodec 58.54.100
 - libavformat 58.29.100                                                                                                                                         - libavutil 56.31.100    
 - installation method: Windows release

![scrpy-d3d-scaling](https://user-images.githubusercontent.com/1296899/82735834-f56a7680-9d24-11ea-8e85-b2ea74487a74.png)

**Comment by rom1v** [comment]
Ref: https://github.com/Genymobile/scrcpy/issues/1394#issuecomment-629345068

**Comment by cjxgm** [comment]
@rustyx Are you sure about `direct3d` being much faster than `opengl`, even on NVIDIA hardware? I know that other vendors do not have good enough `opengl` drivers for Windows (thus `direct3d` support for trilinear filtering is still needed), but NVIDIA's `opengl` driver should be as fast as `direct3d`.

**Comment by rustyx** [comment]
@cjxgm yes, `direct3d` definitely feels more responsive and uses much less CPU, too:
* `--max-fps=20 --render-driver=opengl` : 80% of 1 CPU used
* `--max-fps=20 --render-driver=direct3d`: 1% of 1 CPU used

**Comment by cjxgm** [comment]
Wow that's unexpected and strange. Looks like bug to me (VSync issue?). Should probably open a new issue for that.

**Comment by ChaosEmpire** [comment]
thanks for this thread...

i have had huge issues with 1.16, sessions have bneen so sluggish, i wasnt able to even throw a ball in pokemon remote

i downgraded from 1 16 to 1.14 and got the same error message about direct3d , with 1.16 i didnt got the message, thanks to the error message, i found this thread, and now..i run 1.16 with sdl2 file from 1.12 and its fine

**Comment by JianYuhaha** [comment]
Do you fix the problem? I get it too.

**Comment by rom1v** [comment]
I submitted an issue on SDL few months ago: https://github.com/libsdl-org/SDL/issues/4156
